### ISSUE TITLE
Dux Nutrion pipeline failure

### ISSUE THREAD

**Issue by mariaeduaruda** [dadosfera/Bugsfera#36]
##### There are customers are directly impacted by this bug? Which?
Dux Nutrition

#### Bug Category
- [ ] Connection Manager
- [ ] Connection Test
- [ ] Creating Pipeline
- [x] Executing Pipeline
- [ ] Cataloging Data Asset
- [ ] Other

##### Describe the bug
this pipeline is failing -> [842203a6-de1b-4eb3-a545-1de22aac1853](https://app.dadosfera.ai/pt-BR/collect/pipelines/842203a6-de1b-4eb3-a545-1de22aac1853)

https://app-intelligence-dadosferademo.dadosfera.ai/pbp-service-streamlit-e43fe8e9-5381-4cf1102bb1a7-1370-49c6_8501/Detailed_View

##### Screenshots
![image](https://user-images.githubusercontent.com/87451133/207392206-c9393008-6969-4fe3-b777-24dc889fa60d.png)

##### This bug impact any demo or a sales?
Yes

#### Dadosfera Customer:
Dadosfera

&nbsp;

What environment of software are you using?
- [x] PRD
- [ ] Other

**Comment by beatrizaantunes** [comment]
<img width="859" alt="image" src="https://user-images.githubusercontent.com/79672194/207397126-4e920955-ee5a-47d9-886a-bcbca4ba91b1.png">
@mariaeduaruda @cicerojmm essa pipeline S3 da DUX é aquele que apenas trouxe os dados históricos da VTEX , não foi? Se sim, acho que ela não deveria estar sendo atualizada (na época a gente não tinha agendamento de única extração)

**Comment by rafaelsantanaep** [comment]
@samirleao it seems that the schema has "evolved" and there are more columns in the incremental than in the first version of the table. A workaround would be to redo the full load. There are some solutions to solve this issue permanently:
- Evolve our transpiler to have the complete schema (using the JSON schema in creating the table).
- Have a way to track changes in the schema and add new columns to the table when required.

**Comment by cicerojmm** [comment]
@rafaelsantanaep could this issue be involved by the fact that the s3 connector is now incremental? switched to pipelines that were already active?

**Comment by rafaelsantanaep** [comment]
@cicerojmm Yes, it's happening because it's incremental and the schema `changed`. If it were a full load, it would work because the table was going to be recreated.

**Comment by rafaelsantanaep** [comment]
We made some changes in singer-connector to be able to handle cases when the current `payload` has fewer columns than the table has. I think this would solve this specific problem, but It solves other related issues:
```python
    def auth_options(self):
        return self._create_spark_snowflake_options(self.credentials)

    def _options(self, mode):
        if mode == 'append':
            incremental_options = {
                "column_mismatch_behavior": "IGNORE",
                "column_mapping": "NAME"
            }
            options = self.auth_options()
            options.update(incremental_options)
            return options
        return self.auth_options()

    def save(self, data_frame, mode):
        logger.info(f"Writing table: {self.output_config.table_name} to Snowflake using mode {mode}")
        options = self._options(mode)
        return (
            data_frame.write.format("net.snowflake.spark.snowflake")
            .options(**options)
            .option("dbtable", f"{self.output_config.table_name}")
            .mode(mode)
            .save()
        )
```

**Comment by rafaelsantanaep** [comment]
@cicerojmm I've tested locally the following situation:

1. I added the code above to the s3-connector
2. Created a parquet file and uploaded it to s3. The parquet file had `4 columns` (a, b, c, d).
3. Run the job in incremental mode
4. Added a new parquet file and uploaded it to s3. The parquet file had `5 columns` (a, b, c, d, e).
5. Run the job in incremental mode and it does not fail, but ignored the additional column.

![image](https://user-images.githubusercontent.com/40280691/207634002-21448b5b-bf1b-4148-a085-98ef6b65d18b.png)

**Comment by rafaelsantanaep** [comment]
I think this would be a good workaround to make this job work until the point when we're able to detect changes in the `table_schema` and update it. If we follow in this direction we should document that new columns are ignored and if the new column is required to contact the support.

The support team can refresh the state of the job to force a full refresh that will also include the new columns.

**Comment by samirleao** [comment]
@cicerojmm I breefly discussed this with @rafaelsantanaep and we both agree that to solve this issue ASAP, we could ignore new columns during write to Snowflake, or simply reset the incremental state to let it reload all the historical data. The most definitive solution would be to interact with the catalog containing the table schema before the write operation, and manage the schema evolution (create the new column with default NULL values with alter table before the write operation, and further register the schema change on the catalog).

**Comment by rafaelsantanaep** [comment]
Closed deployed by [#54](https://github.com/dadosfera/s3-connector/pull/54)

Tested the job that was failing and it worked.

![image](https://user-images.githubusercontent.com/40280691/207749938-7cc1e150-6c03-49e1-970f-ef695f66c14e.png)
